packages/analytics-core/src/plugins/destination.ts

```diff
@@ -59,16 +59,16 @@ export class Destination implements DestinationPlugin {
     });
   }
 
-  async addToQueue(...list: Context[]): Promise<void> {
-    const tryable = list.filter(async (context) => {
+  addToQueue(...list: Context[]) {
+    const tryable = list.filter((context) => {
```

Comment:
i reverted the async stuff here. the integration tests had inconsistent results when running single/multiple test and when reordered. i finally figured it was due to a test that exhausts maximum retry limit. the issue is when list.filter() is passed with an async callback, the callback always return a truthy value and therefore does not perform filter correctly. this lead to the test continuously retrying and ignoring the max retry limit as checked on the line after this